server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/DatafileManager.java

```diff
@@ -571,4 +583,19 @@ public MetadataUpdateCount getUpdateCount() {
     return metadataUpdateCount.get();
   }
 
+  public void logTransactions() {
+    if (tabletLog.getNumTransactions() > 0) {
+      log.error("Operation log: {}", tabletLog.dumpLog());
+    }
+  }
+
+  public void checkTransactionLog() {
+    if (tabletLog.getNumTransactions() > 0) {
+      Set<StoredTabletFile> files = datafileSizes.keySet();
+      if (!tabletLog.isExpectedFiles(files)) {
+        log.error("In-memory files {} do not match transaction log {}", files);
+        logTransactions();
+      }
+    }
+  }
```

Comment:
These should probably still run even when the size is zero.

```suggestion
  public void logTransactions() {
     log.error("Operation log: {}", tabletLog.dumpLog());
  }

  public void checkTransactionLog() {
      Set<StoredTabletFile> files = datafileSizes.keySet();
      if (!tabletLog.isExpectedFiles(files)) {
        log.error("In-memory files {} do not match transaction log {}", files);
        logTransactions();
      }
  }
```

Reply:
ok, looking at that.

Reply:
done.